src/os/bluestore/KernelDevice.cc

```diff
@@ -551,6 +551,39 @@ int KernelDevice::read(uint64_t off, uint64_t len, bufferlist *pbl,
   return r < 0 ? r : 0;
 }
 
+int KernelDevice::aio_read(
+  uint64_t off,
+  uint64_t len,
+  bufferlist *pbl,
+  IOContext *ioc)
+{
+  dout(5) << __func__ << " 0x" << std::hex << off << "~" << len << std::dec
+	  << dendl;
+
+  int r = 0;
+#ifdef HAVE_LIBAIO
+  if (aio && dio) {
+    _aio_log_start(ioc, off, len);
+    ioc->pending_aios.push_back(FS::aio_t(ioc, fd_direct));
+    ++ioc->num_pending;
+    FS::aio_t& aio = ioc->pending_aios.back();
```

Comment:
`emplace_back` returns `void` in [cpp11, cpp17), and returns `referent` since cpp17. FYI http://en.cppreference.com/w/cpp/container/vector/emplace_back